### PR TITLE
Fix to handle interval custom cron (#6745)

### PR DESCRIPTION
**Pull request by swadeley**
Fix test_positive_create_with_interval for custom cron #6745

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanCreateTestCase::test_positive_create_with_interval

======================== test session starts =============================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-15 12:57:43 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                             

tests/foreman/api/test_syncplan.py .                                                              [100%]

======================= 1 passed in 4.12 seconds ===========================

File: tests/foreman/api/test_syncplan.py [hunk 1]
```diff
@@ -193,11 +193,21 @@ def test_positive_create_with_interval(self):
...
-            with self.subTest(interval):
+            if interval != SYNC_INTERVAL['custom']:
                 sync_plan = entities.SyncPlan(
-                    interval=interval,
+                     description=gen_string('alpha'),
+                     organization=self.org,
+                     interval=interval
+                ).create()
+
+            elif (interval == 'custom cron'):
```
Comment (ldjebran) [inline tests/foreman/api/test_syncplan.py]: simply else, as if iterval != SYNC_INTERVAL['custom'] the else case automatically should be equal
any way prefere to invert the condition: 
eg:
```python
if interval == SYNC_INTERVAL['custom']:
    ...
else:
    ...

```
what do you think?

Comment (pondrejk) [reply tests/foreman/api/test_syncplan.py]: I agree that the second condition is redundant

File: tests/foreman/api/test_syncplan.py [hunk 2]
```diff
@@ -193,11 +193,21 @@ def test_positive_create_with_interval(self):
...
+                sync_plan = entities.SyncPlan(
+                    description=gen_string('alpha'),
                     organization=self.org,
+                    interval='custom cron',
```
Comment (pondrejk) [inline tests/foreman/api/test_syncplan.py]: here we should better use the variable `interval=interval,` as above, if the value (custom cron) changes wording in the future, we would need to update it also here

File: tests/foreman/api/test_syncplan.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/api/test_syncplan.py]: Thanks for the review. I will invert the condition.

File: tests/foreman/api/test_syncplan.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/api/test_syncplan.py]: OK, commit coming.

**Comment by swadeley** [comment]
robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanCreateTestCase::test_positive_create_with_interval
==================== test session starts =================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-19 12:28:41 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                             

tests/foreman/api/test_syncplan.py .                                                              [100%]

=================== 1 passed in 4.63 seconds =========================

 robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanCreateTestCase
====================== test session starts ===============================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-19 12:55:06 - conftest - DEBUG - BZ deselect is disabled in settings

collected 8 items                                                                                                            

tests/foreman/api/test_syncplan.py ........                                                             [100%]

===================== 8 passed in 8.94 seconds ============================

File: tests/foreman/api/test_syncplan.py [hunk 3]
```diff
@@ -193,11 +193,21 @@ def test_positive_create_with_interval(self):
...
                 ).create()
+
+            else:
+                sync_plan = entities.SyncPlan(
+                     description=gen_string('alpha'),
+                     organization=self.org,
+                     interval=interval
+                ).create()
+
                 self.assertEqual(sync_plan.interval, interval)
```
Comment (mirzal) [inline tests/foreman/api/test_syncplan.py]: This assertion will run only for non-custom intervals. For custom interval, we create Sync Plan and assume that if it didn't throw error, everything is OK.
Is that intentional?

**Comment by mirzal** [comment]
@swadeley You don't have to create entity in both conditional paths. You can just add field to existing entity conditionally. Something like that:

```
sync_plan = entities.SyncPlan(
    description=gen_string('alpha'),
    organization=self.org,
    interval=interval
)
if interval == SYNC_INTERVAL['custom']:
    sync_plan.cron_expression = gen_choice((valid_cron_expressions()))
sync_plan = sync_plan.create()
```

File: tests/foreman/api/test_syncplan.py (hunk 3 above)
Comment (swadeley) [reply tests/foreman/api/test_syncplan.py]: No, that is a mistake. Thanks for noticing.

**Comment by swadeley** [comment]
@mirzal I will try that. Thank you.

**Comment by swadeley** [comment]
robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanCreateTestCase::test_positive_create_with_interval
====================== test session starts =========================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-20 11:52:57 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                             

tests/foreman/api/test_syncplan.py .                                                        [100%]

==================== 1 passed in 6.75 seconds ===========================

robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanCreateTestCase
======================== test session starts ==============================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-20 11:57:04 - conftest - DEBUG - BZ deselect is disabled in settings

collected 8 items                                                                                                            

tests/foreman/api/test_syncplan.py ........                                                       [100%]

====================== 8 passed in 9.10 seconds =============================

**Comment by swadeley** [comment]
@mirzal Hello, please review once more.
Thank you

**Comment by mirzal** [comment]
@swadeley can you move entire loop body one indentation level up?
Right now they are indented by 8 spaces (relative to `for`), while we usually use 4 spaces for indent. [PEP 8 also recommends 4 spaces](https://www.python.org/dev/peps/pep-0008/#indentation).

Other than that, this looks great :+1:

**Comment by swadeley** [comment]
@mirzal Change made, thank you.

**Pull request merged**